cmd/ops_agent_uap_plugin/service_linux.go

```diff
@@ -130,6 +157,103 @@ func (ps *OpsAgentPluginServer) GetStatus(ctx context.Context, msg *pb.GetStatus
 	return &pb.Status{Code: 0, Results: []string{"The Ops Agent Plugin is running ok."}}, nil
 }
 
+// runSubagents starts up the diagnostics service, otel, and fluent bit subagents in separate goroutines.
+// All child goroutines create a new context derived from the same parent context.
+// This ensures that crashes in one goroutine don't affect other goroutines.
+// However, when one goroutine exits with errors, it won't be restarted, and all other goroutines are also terminated.
+// This is done by canceling the parent context.
+// This makes sure that GetStatus() returns a non-healthy status, signaling UAP to Start() the plugin again.
+//
+// ctx: the parent context that all child goroutines share.
+//
+// cancel: the cancel function for the parent context. By calling this function, the parent context is canceled,
+// and GetStatus() returns a non-healthy status, signaling UAP to re-trigger Start().
+func runSubagents(ctx context.Context, cancel context.CancelFunc, pluginInstallDirectory string, pluginStateDirectory string, restartCommand RestartCommandFunc, runCommand RunCommandFunc) {
+	// Register signal handler and implements its callback.
+	sigHandler(ctx, func(_ os.Signal) {
+		cancel()
+	})
+
+	var wg sync.WaitGroup
+	// Starting the diagnostics service
+	runDiagnosticsCmd := exec.CommandContext(ctx,
+		path.Join(pluginInstallDirectory, DiagnosticsBinary),
+		"-config", OpsAgentConfigLocationLinux,
+	)
+	wg.Add(1)
+	go restartCommand(ctx, cancel, runDiagnosticsCmd, runCommand, &wg)
+
+	// Starting Otel
+	runOtelCmd := exec.CommandContext(ctx,
+		path.Join(pluginInstallDirectory, OtelBinary),
+		"--config", path.Join(pluginStateDirectory, OtelRuntimeDirectory, "otel.yaml"),
+	)
+	wg.Add(1)
+	go restartCommand(ctx, cancel, runOtelCmd, runCommand, &wg)
+
+	// Starting FluentBit
+	runFluentBitCmd := exec.CommandContext(ctx,
+		path.Join(pluginInstallDirectory, AgentWrapperBinary),
+		"-config_path", OpsAgentConfigLocationLinux,
+		"-log_path", path.Join(pluginStateDirectory, LogsDirectory, "subagents/logging-module.log"),
+		path.Join(pluginInstallDirectory, FluentbitBinary),
+		"--config", path.Join(pluginStateDirectory, FluentBitRuntimeDirectory, "fluent_bit_main.conf"),
+		"--parser", path.Join(pluginStateDirectory, FluentBitRuntimeDirectory, "fluent_bit_parser.conf"),
+		"--storage_path", path.Join(pluginStateDirectory, FluentBitStateDiectory, "buffers"),
+	)
+	wg.Add(1)
+	go restartCommand(ctx, cancel, runFluentBitCmd, runCommand, &wg)
+
+	wg.Wait()
+}
+
+func restartCommand(ctx context.Context, cancel context.CancelFunc, cmd *exec.Cmd, runCommand RunCommandFunc, wg *sync.WaitGroup) {
```

Comment:
Should this still be called `restartCommand` now that we don't actually restart the subagents after a crash?